lumigator/python/mzai/backend/backend/services/jobs.py

```diff
@@ -284,7 +288,17 @@ def get_job_result(self, job_id: UUID) -> JobResultResponse:
     def get_job_result_download(self, job_id: UUID) -> JobResultDownloadResponse:
         """Return job results file URL for downloading."""
         # Generate presigned download URL for the object
-        result_key = self._get_results_s3_key(job_id)
+        job_info = self.ray_client.get_job_info(job_id)
```

Comment:
A heads up that one might want to download a job result even after a system restart.
Data about experiments _should_ be persisted in the backend (DB + S3) but it would likely not persist in ray (especially in the local setup).

I think the main problem right now is that we are changing the output results filename depending on the job type. This is not really necessary as we are not planning a single job to return more than one output file.

Do you think it would make sense to just call the file `results.json` regardless of the job type? This would allow download to be agnostic of the job type (as it should be). If we want to introduce any more flexibility, I think storing S3 path together with job metadata would help, but I think it is out of scope for the current PR.

Reply:
That's even simpler. We can revisit it at some point, if we need a job to create more than one result files. I'll keep the commit which passes the job type through the metadata structure, and simplify the rest.

Reply:
Thanks for keeping the metadata commit, I think it will come handy sooner or later! 🙏